lib/carin_for_blue_button_test_kit/docs/carin_for_blue_button_v2.0.0_client_suite_description.md

```diff
@@ -0,0 +1,106 @@
+  - Be able to process and retain all profile data elements that have a minimum cardinality >= 1 and/or flagged
+  as Must Support as defined by that profiles StructureDefinition
+  - Conform to the US Core Client Capability Statement expectations for that profiles type
+
+## Scope
+
+These tests are a **DRAFT** intended to allow implementers to perform
+preliminary checks of their systems against the requirements stated for CARIN IG for Blue Button client actors
+and [provide feedback](https://github.com/inferno-framework/subscriptions-test-kit/issues)
+on the tests. Future versions of these tests may verify other
+requirements and may change the test verification logic.
+
+## Test Methodology
+
+For these tests Inferno simulates a Carin for Blue Button server application. Inferno will wait for
+the client under test to request a resource for each of the Carin for Blue Button profiles, which include the Patient,
+Coverage, Inpatient Institutional ExplanationOfBenefit, Oral ExplanationOfBenefit, Outpatient Institutional ExplanationOfBenefit,
+Pharmacy ExplanationOfBenefit, Professional NonClinician ExplanationOfBenefit, Organization, Practitioner, and RelatedPerson profiles.
+The tests also waits for search requests to be made for each of the required searches described for each resource in the
+Carin for Blue Button IG and the US Core 3.1.1 IG. For each request sent in from the client, Inferno will forward the
+request to the Inferno Reference Server, where it will grab the appropriate Carin for Blue Button resources to send back to the
+client. Each of these resources will contain all of the required and Must Support elements specified by the Carin for Blue Button IG
+to ensure the client can process a resource containing all of these fields.
+
+Once the client has finished submitting requests, Inferno will verify that the client has requested and received the correct
+resources from the Inferno Reference server for each of the required Carin for Blue Button profiles, and it will verify that a
+search request was made for each of the required searches listed for each resource. Using the interactions between Inferno and the
+client under test, Inferno will verify that the client meets the obligations from the implementation guide by ensuring the client is
+able to request and receive all the required resource profiles.
+
+## Running the Tests
+
+### Quick Start
+
+The following input must be provided by the tester to execute
+any tests in this suite:
+1. *Access Token*: A `Bearer` token that the client under test will send in the 
+   `Authorization` header of HTTP requests made against Inferno. Inferno uses the
+   value to identify incoming requests that belong to the testing session.
+
+Once the testing starts, Inferno will wait for Carin for Blue Button resource and search requests from the client,
+forward these requests to the Inferno Reference Server, then return the Reference Server's response to
+the client and verify the interaction.
```

Comment:
Leave out the detail on the Inferno Reference Server.

One question that isn't clear is how the patient is coordinated. I think right now we are assuming that the client under test knows the specific patient Id that is stored on the reference server and that these tests are making available. That might be what we need to keep doing for this release, but I don't think its a viable long-terms strategy.  Is there anything in the CARIN IG that talks about patient identification or patient searches that might direct us on how to approach this?